### PR TITLE
Test/poll controller

### PR DESCRIPTION
**Pull request by Severinzz**
**Added**
* Test for PollController
* Admin check for getAllPolls
* Return false when there is no start time for poll (fix nullpointer execption)

**Unsolved issues**
* When persisting vote without setting the ID manually there is an [InvalidDataAccessApiUsageException](https://gyazo.com/fca36841b8640a21846f6626bee23d25). When setting the voteId manually there is no error.

**Note**
* Test for "admin can get all polls by poll id" not made since as the application stands everyone can get the poll as long as they have the id for the poll. Test not necessary when we are not keeping track of who are "invited" to vote on the poll/have a list of users that is allowed to access that poll. In my oppinion

### CODE REVIEW

File: src/test/java/no/hvl/dat250/h2020/group5/controllers/PollControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+        Poll poll = response.getBody();
+
+        Assertions.assertEquals(poll1, poll);
+    }
+
+    @Test
+    public void shouldDeletePollByPollId() {
+        template.delete(base.toString() + "/" + poll1.getId().toString() + "/" + user1.getId().toString());
+        Assertions.assertTrue(pollRepository.findById(poll1.getId()).isEmpty());
+    }
```
Comment (oddhus) [inline src/test/java/no/hvl/dat250/h2020/group5/controllers/PollControllerTest.java]: Burde vi ha en test som også sjekker om stemmene er blitt slettet?

File: src/main/java/no/hvl/dat250/h2020/group5/service/PollService.java [hunk 2]
```diff
@@ -65,10 +65,13 @@ public boolean deletePoll(Long pollId, Long userId) {
                 .collect(Collectors.toList());
     }
 
-    public List<PollResponse> getAllPolls() {
-        return pollRepository.findAll().stream()
-                .map(PollResponse::new)
-                .collect(Collectors.toList());
+    public List<PollResponse> getAllPolls(Long userId) {
+        if (isAdmin(userId)) {
```
Comment (oddhus) [inline src/main/java/no/hvl/dat250/h2020/group5/service/PollService.java]: Flott at du la til en admin sjekk!

**Review by oddhus — DISMISSED** [review]
Gode og dekkkende tester! 🥇

**Comment by oddhus** [comment]
> **Unsolved issues**
> When persisting vote without setting the ID manually there is an InvalidDataAccessApiUsageException. When setting the voteId manually there is no error.

Jeg lager en issue på dette

**Review by oddhus — DISMISSED** [review]

File: src/main/java/no/hvl/dat250/h2020/group5/service/PollService.java [hunk 3]
```diff
@@ -116,6 +116,10 @@ public boolean isActivated(Long pollId) {
             return false;
         }
 
+        if (poll.get().getStartTime() == null) {
+            return false;
+        }
+
```
Comment (JennStro) [inline src/main/java/no/hvl/dat250/h2020/group5/service/PollService.java:122]: Bra at du sjekker dette! 👍

File: src/test/java/no/hvl/dat250/h2020/group5/controllers/PollControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        pollRepository.deleteAll();
+        userRepository.deleteAll();
+        guestRepository.deleteAll();
+
+        user1 = new User();
+        user1.setUsername("Admin");
+        user1.setIsAdmin(true);
```
Comment (JennStro) [inline src/test/java/no/hvl/dat250/h2020/group5/controllers/PollControllerTest.java:55]: Hvis du vil så kan dette bli forenklet til 
```suggestion
        user1 = new User().userName("Admin").admin(true);
```

:)

File: src/test/java/no/hvl/dat250/h2020/group5/controllers/PollControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+        User user2 = new User();
+        user2.setUsername("Not admin");
+
+        Guest guest1 = new Guest();
+        guest1.setUsername("guest1");
+
+        poll1 = new Poll();
+        poll1.setQuestion("Question");
+        poll1.setPollOwner(user1);
+        poll1.setVisibilityType(PollVisibilityType.PUBLIC);
```
Comment (JennStro) [inline src/test/java/no/hvl/dat250/h2020/group5/controllers/PollControllerTest.java:66]: Samme som i Guest. Hvis du gidder kan du lage en metode i Poll `public Poll question(...) {setQuestion(question); return this;}` og erstatte med 
```suggestion
        poll1 = new Poll().question("Question").pollOwner(user1).visibilityTyoe(PollVisibilityType.PUBLIC);
```

File: src/test/java/no/hvl/dat250/h2020/group5/controllers/PollControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+        poll1 = pollRepository.save(poll1);
+        pollRepository.save(poll2);
+        user1 = userRepository.save(user1);
+        guestRepository.save(guest1);
+
+        Vote vote1 = new Vote();
+        vote1.setVoter(user1);
+        vote1.setPoll(poll1);
+        vote1.setAnswer(AnswerType.NO);
+        vote1.setId((long) 123123);
```
Comment (JennStro) [inline src/test/java/no/hvl/dat250/h2020/group5/controllers/PollControllerTest.java:82]: Kanskje vi bør vurdere å lage builder-pattern til alle entitietene, slik at det blir lettere å instansiere de :) (slik som User og Poll)

**Review by JennStro — APPROVED** [review]
Kjempeflotte tester, bra jobbet! 🔥 🔥

**Pull request merged**